### PR TITLE
Add generic support to retry failing tests

### PR DESCRIPTION
**Pull request by NotTheEvilOne**
**What this PR does / why we need it**:
This PR adds support to add a wrapper around tests to retry them for a specified amount of times if a given list of exceptions occurs. This is used for `base/test/test_time_config.py::test_ntp` and may be added for other tests as required.

### CODE REVIEW

File: tests/helper/wrappers.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from asyncio import sleep as asyncio_sleep
+from functools import wraps
+from time import sleep
+
+async def async_retry(py_callable, retries = 3, wait_time = 3, exceptions_retried = ( AssertionError, )):
```
Comment (yeoldegrove) [inline tests/helper/wrappers.py:5]: To make it even more robust initially, we could wait for 10s before retrying again. What do you think?
Otherwise, I like this feature and that it can be chosen on a per test basis.

**Review by yeoldegrove — CHANGES_REQUESTED** [review]

File: tests/helper/wrappers.py (hunk 1 above)
Comment (moritzkirchnersap) [reply tests/helper/wrappers.py:5]: While I like the functionality I'm wondering if there is a pytest package with similar functionality. If there are packages then I'm always a proponent of using them because my naive assumption is they work.

What do we think about pytest-rerunfailures https://pypi.org/project/pytest-rerunfailures/#re-run-all-failures-matching-certain-expressions

which should allow something like:
`@pytest.mark.flaky(rerun_except="AssertionError", reruns=5, reruns_delay=2)`

Comment (NotTheEvilOne) [reply tests/helper/wrappers.py:5]: We can increase the default value to e.g. 5 seconds, 10 might be a bit high. You can always customize it if needed per test basis @yeoldegrove.

`@pytest.mark.flaky` seems to contain all features implemented here as well, you are right @moritzkirchnersap :)

**Review by yeoldegrove — APPROVED** [review]
Thanks for this feature. Approved.

Few minor things:
- Does it make sense to document somewhere that we can re-run tests this way? I am not sure what we have documented around pytest, yet ...
- The version bump is because we use a newer python version since a few nighties, right? Did you test that this works?

**Comment by NotTheEvilOne** [comment]
[...]
>     * The version bump is because we use a newer python version since a few nighties, right? Did you test that this works?

Yes, as I touched Pipfile anyway I updated Python as well. Most tests are done locally. As we have it in `platform-test-image`s I assume that no corner cases hide there.
